### PR TITLE
Import more of results.json into the database

### PR DESCRIPTION
**Pull request by coolo**
I had to squash several commits into one to avoid a huge merge mess (as
the whole branch took too much time):

- parse flags into JobModules
- remove unused test_branch from Jobs
- add backend_infos to Jobs
- various fixes for /tests route
- transfer results.json in status call to get info about running job

### CODE REVIEW

**Comment by coolo** [comment]
This is still incomplete, but I would like to have this merged into master and fix the remainings there. The test suite passes and my testing was successfull too.

os-autoinst needs to split the results.json correctly. there is no point in writing migrations doing that if os-autoinst will keep writing huge JSON files.

File: lib/OpenQA/Controller/Worker.pm [hunk 1]
```diff
@@ -44,11 +44,11 @@ sub worker_info($) {
     # puts job id in status, otherwise is idle
     if($job) {
         my $testdirname = $job->{'settings'}->{'NAME'};
-        my $results = test_result($testdirname);
-        my $modinfo = get_running_modinfo($results);
         $settings->{status} = "running";
         $settings->{jobid} = $job->{id};
-        $settings->{currentstep} = $modinfo->{running};
+        my $schema = Scheduler::schema();
```
Comment (aaannz) [inline lib/OpenQA/Controller/Worker.pm]: Here should be OpenQA::Scheduler::schema

I just noticed 'lib/OpenQA' is still put into @INC in compile check so that's probably why this passed that test

**Comment by coolo** [comment]
no, perl doesn't care at compile time ;(

Fixed it.

File: lib/OpenQA/Utils.pm [hunk 2]
```diff
@@ -20,11 +20,9 @@ $VERSION = sprintf "%d.%03d", q$Revision: 1.12 $ =~ /(\d+)/g;
   &first_run
   &data_name
   &back_log
-  &get_running_modinfo
   &needle_info
   &needledir
   &testcasedir
-  &test_result
   &test_result_module
```
Comment (aaannz) [inline lib/OpenQA/Utils.pm]: This should be removed too since you removed the function

File: lib/OpenQA/Controller/Running.pm [hunk 3]
```diff
@@ -57,8 +58,7 @@ sub modlist {
     my $self = shift;
     return 0 unless $self->init();
 
-    my $results = test_result($self->stash('testdirname'));
-    my $modinfo = get_running_modinfo($results);
+    my $modinfo = Schema::Result::JobModules::running_modinfo($self->stash('job'));
```
Comment (aaannz) [inline lib/OpenQA/Controller/Running.pm]: Sorry, this one I missed in the first run. OpenQA:: prefix missing

**Comment by aaannz** [comment]
As far as I can tell LGTM, given that its WIP. There are some prints and Dumper outputs we don't want in final I suppose.

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 4]
```diff
@@ -76,10 +76,6 @@ __PACKAGE__->add_columns(
     test => {
         data_type => 'text',
     },
-    test_branch => {
-        data_type => 'text',
-        is_nullable => 1,
-    },
```
Comment (lnussel) [inline lib/OpenQA/Schema/Result/Jobs.pm]: seems to be unrelated and doesn't even if still unused so why remove?

Comment (coolo) [inline lib/OpenQA/Schema/Result/Jobs.pm]: as I reviewed the code, I cleaned it up

File: lib/OpenQA/Controller/Step.pm [hunk 5]
```diff
@@ -335,14 +323,15 @@ sub edit {
...
-    my $results = $self->stash('results');
+    my $job = $self->stash('job');
     my $module = $self->stash('module');
 
-    my $testcasedir = testcasedir($results->{distribution}, $results->{version});
-    my $scriptpath = "$testcasedir/$module->{'script'}";
+    my $testcasedir = testcasedir($job->{settings}->{DISTRI}, $job->{settings}->{VERSION});
+    my $scriptpath = "$testcasedir/" . $module->script;
+    print "S $scriptpath\n";
     if(!$scriptpath || !-e $scriptpath) {
```
Comment (lnussel) [inline lib/OpenQA/Controller/Step.pm]: extra print

File: lib/OpenQA/Controller/Step.pm [hunk 6]
```diff
@@ -525,30 +513,36 @@ sub calc_min_similarity($$) {
 sub viewimg {
     my $self = shift;
     my $module_detail = $self->stash('module_detail');
-    my $results = $self->stash('results');
+    my $job = $self->stash('job');
+    my $distribution = $job->{settings}->{DISTRI};
+    my $dversion = $job->{settings}->{VERSION} || '';
+
+    print Dumper($module_detail);
```
Comment (lnussel) [inline lib/OpenQA/Controller/Step.pm]: extra print

File: lib/OpenQA/Scheduler.pm [hunk 7]
```diff
@@ -782,9 +783,11 @@ sub job_update_status($$) {
     my ($id, $status) = @_;
 
     my $job = _job_get({ 'me.id' => $id });
-    #    print "$id " . Dumper($status) . "\n";
+    # print "$id " . Dumper($status) . "\n";
 
     _append_log($job, $status->{log});
+
+    Schema::Result::JobModules::split_results($job, $status->{results});
 }
```
Comment (coolo) [inline lib/OpenQA/Scheduler.pm]: one more oversight :)
